tools/api_boost/api_boost.py

```diff
@@ -0,0 +1,161 @@
+    if include_lines and line.startswith('#include'):
+      output_lines.extend(include_lines)
+      include_lines = None
+    # Exclude API includes, except for a special case related to v2alpha
+    # ext_authz; this is needed to include the service descriptor in the build
+    # and is a hack that will go away when we remove v2.
+    if re.match(API_INCLUDE_REGEX, line) and 'envoy/service/auth/v2alpha' not in line:
+      continue
+    output_lines.append(line)
+
+  # Write to temporary file. We can't overwrite in place as we're executing
+  # concurrently with other ApiBoostFile() invocations that might need the file
+  # we're writing to.
+  pathlib.Path(path + TMP_SWP_SUFFIX).write_text('\n'.join(output_lines) + '\n')
+
+
+# Replace the original file with the temporary file created by ApiBoostFile()
+# for a given path.
+def SwapTmpFile(path):
+  pathlib.Path(path + TMP_SWP_SUFFIX).rename(path)
+
+
+# Update the Envoy source tree the latest API.
+def ApiBoostTree(args):
+  # Optional setup of state. We need the compilation database and api_booster
+  # tool in place before we can start boosting.
+  if args.generate_compilation_database:
+    sp.run(['./tools/gen_compilation_database.py', '--run_bazel_build', '--include_headers'],
+           check=True)
+
+  if args.build_api_booster:
+    # Similar to gen_compilation_database.py, we only need the cc_library for
+    # setup. The long term fix for this is in
+    # https://github.com/bazelbuild/bazel/issues/9578.
+    dep_build_targets = [
+        '//source/...',
+        '//test/...',
+    ]
+    # Figure out some cc_libraries that cover most of our external deps. This is
+    # the same logic as in gen_compilation_database.py.
+    query = 'attr(include_prefix, ".+", kind(cc_library, deps({})))'.format(
+        ' union '.join(dep_build_targets))
+    dep_lib_build_targets = sp.check_output(['bazel', 'query', query]).decode().splitlines()
+    # We also need some misc. stuff such as test binaries for setup of benchmark
+    # dep.
+    query = 'attr("tags", "compilation_db_dep", {})'.format(' union '.join(dep_build_targets))
+    dep_lib_build_targets.extend(sp.check_output(['bazel', 'query', query]).decode().splitlines())
+    extra_api_booster_args = []
+    if args.debug_log:
+      extra_api_booster_args.append('--copt=-DENABLE_DEBUG_LOG')
+
+    # Slightly easier to debug when we build api_booster on its own.
+    sp.run([
+        'bazel',
+        'build',
+        '--strip=always',
+        '@envoy_dev//clang_tools/api_booster',
+    ] + extra_api_booster_args,
+           check=True)
+    sp.run([
+        'bazel',
+        'build',
+        '--strip=always',
+    ] + dep_lib_build_targets, check=True)
+
+  # Figure out where the LLVM include path is. We need to provide this
+  # explicitly as the api_booster is built inside the Bazel cache and doesn't
+  # know about this path.
+  # TODO(htuch): this is fragile and depends on Clang version, should figure out
```

Comment:
from the previous PR, what does we actually include from versioned include?

Reply:
Copy+pasting:

If I don't declare this as `-isystem`, I see:

```
...clang-9.0/lib/clang/9.0.0/include/mmintrin.h:50:12: error: use of old-style cast [-Werror,-Wold-style-cast]
    return (__m64)__builtin_ia32_vec_init_v2si(__i, 0);    
```

since it doesn't recognize these libraries as system libs (and hence subject to relaxation of the warning).